staging/src/k8s.io/client-go/tools/portforward/tunneling_connection.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package portforward
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	gwebsocket "github.com/gorilla/websocket"
+
+	"k8s.io/klog/v2"
+)
+
+const writeDeadline = 2 * time.Second
```

Comment:
how was this deadline selected? I'm not sure I'd expect us to hard code a write deadline like this

Reply:
This is the same deadline as in the `RemoteCommand` code. The comments are:
```
// writeDeadline defines the time that a write to the websocket connection                                                                                        
// must complete by, otherwise an i/o timeout occurs. The writeDeadline                                                                                           
// has nothing to do with a response from the other websocket connection                                                                                          
// endpoint; only that the message was successfully processed by the                                                                                              
// local websocket connection. The typical write deadline within the websocket                                                                                    
// library is one second.                                                                                                                                         
const writeDeadline = 2 * time.Second     
```

Reply:
since we're only using this in the close connection now, inline this and shorten it to 1 second

separately, I think we need to revisit the RemoteCommand use of this and make sure we're not making the websocket version more vulnerable to network stalls than the spdy version

Reply:
Inlined and shortened write deadline to 1 second for this close signal.